app/domain/people/household_list.rb

```diff
@@ -10,160 +10,104 @@ class People::HouseholdList
 
   include Enumerable
 
-  def initialize(people_scope)
+  def initialize(people_scope, retain_order: false, include_housemates: false)
     @people_scope = people_scope
+    @retain_order = retain_order
+    @include_housemates = include_housemates
   end
 
-  def only_households_in_batches(&block)
-    return unless block
-
-    fetch_in_batches(only_households, &block)
+  def only_households_in_batches(&)
+    in_batches(@people_scope.where.not(household_key: nil), &)
   end
 
-  def people_without_household_in_batches(&block)
-    return unless block
+  def people_without_household_in_batches(&)
+    in_batches(@people_scope.where(household_key: nil), &)
+  end
 
-    fetch_in_batches(people_without_household, &block)
+  def households_in_batches(&)
+    in_batches(@people_scope, &)
   end
 
-  def households_in_batches(&block)
-    return unless block
+  def count
+    return [] if @people_scope.none?
 
-    fetch_in_batches(grouped_households, &block)
+    person_ids_grouped_by_household_query(@people_scope).to_a.size
   end
 
-  def grouped_households
-    Person
-      .select("people.*")
-      .from(grouped_households_people_sql, :people)
-      .limit(@people_scope.limit_value.presence)
-  end
```

Comment:
Diese interne Methode habe ich entfernt. Dass die intern ist behaupte ich einfach so, da müsste man sicher noch in den Wagons checken, ob die sonst gebraucht wird.